### PR TITLE
#160370654 Enable Managers to See Full Details of Multi-city Requests

### PR DESCRIPTION
**Pull request by johnmutuma5**
#### What does this PR do?
- Enables managers to see full details of trips within a multi-city travel request for approval

#### Description of Task to be completed?
Managers need to see full details of the trips within each request so that there is sufficient information for approval decisions

#### How should this be manually tested?
- Clone this repo - `git clone https://github.com/andela/travel_tool_back.git`
- Change into the repo directory - `cd travel_tool_back`
- Check out this branch - `git checkout ft-approve-multicity-requests-160370654`
- Install dependencies and start the server - `make start`
- Using Postman:
  - Set your token in the Authorization Header - you may need to get an authorization token by login in with a browser and retrieving the token from the browser's developer tools in the Application tab, Session cookies section
  - Send a GET request to `http://localhost:5000/api/v1/requests/<reqeustId>` to retrieve the details of a request:
    - Checkout background context below to see how you can get a testable `requestId`
     

#### Any background context you want to provide?
To get details of a request, you'll need a `requestId`. This is autogenerated in the backend when a new request is created.
- Use Postman to send a POST request to http://localhost:5000/api/v1/requests/ with the data in [this](https://gist.github.com/johnmutuma5/45cd66d793c52319bde97731be994ec5) public gist as the request body
Remember to change the "manager" value in the .json to be your name as it appears in your google account
- Once you get a success response, you can locate a `requestId` within the response body and utilize it in the testing section above

#### What are the relevant pivotal tracker stories?
[#160370654](https://www.pivotaltracker.com/story/show/160370654)

#### Screenshots (if appropriate)
screen shot 2018-09-17 at 10 19 55 am

### CODE REVIEW

**Comment by Collins33** [comment]
@johnmutuma5 You have not added the relevant pivotal tracker story ID

**Comment by johnmutuma5** [comment]
Thanks @Collins33, done.

**Comment by Collins33** [comment]
@johnmutuma5 on the part for testing the PR, are we cloning the backend or the front-end? I need some clarification on this.

**Comment by johnmutuma5** [comment]
@Collins33, updated the description.

File: src/database/models/Approval.js [hunk 1]
```diff
@@ -6,26 +6,6 @@ export default (sequelize, DataTypes) => {
       autoIncrement: true,
       type: DataTypes.INTEGER
     },
-    requestId: {
```
Comment (d3mola) [inline src/database/models/Approval.js]: @johnmutuma5 please why are we removing this part from the Approvals models?

File: src/database/models/trip.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+export default (sequelize, DataTypes) => {
```
Comment (d3mola) [inline src/database/models/trip.js]: @johnmutuma5, from the feedback I got from @samuelkubai on my PR yesterday, this file name should in Title case

**Comment by d3mola** [comment]
@johnmutuma5 the title of your PR sort of conflicts with the section of the description that says `What does this PR do?`. The title tells me that you're approving multi-city requests while the description tells me you're making sure managers can see trips associated with a multi-city request.

**Comment by d3mola** [comment]
@johnmutuma5 when I try to post a new request and I enter a `tripType` that does not exist(i.e. not one of `oneWay`, `multi` and `return`), I get a sequelize database error(since the tripType I added is not one of the allowed values in the db), Ideally, there should be a validator that makes sure the invalid request doesn't even hit the database.

I'm not sure if this is out of the scope of your work since your task is not concerned with creating requests but just getting the details of that request. Could you sync with @amarachukwu-agbo since she worked on posting? I just need clarity

File: src/database/models/trip.js (hunk 2 above)
Comment (johnmutuma5) [reply src/database/models/trip.js]: Thanks @d3mola, Updating that.

**Comment by johnmutuma5** [comment]
> @johnmutuma5 when I try to post a new request and I enter a `tripType` that does not exist(i.e. not one of `oneWay`, `multi` and `return`), I get a sequelize database error(since the tripType I added is not one of the allowed values in the db), Ideally, there should be a validator that makes sure the invalid request doesn't even hit the database.
> 
> I'm not sure if this is out of the scope of your work since your task is not concerned with creating requests but just getting the details of that request. Could you sync with @amarachukwu-agbo since she worked on posting? I just need clarity

@d3mola, your description is definitely the ideal case. @seunzone, @amarachukwu-agbo and @Spectrumsun  will be working on creating the full logic of creating these types of requests in the database. What I put there was a temporary one-liner in the `RequestsController` for demonstration purposes because you could not retrieve a request with trips without being able to create one in the first place.

**Comment by johnmutuma5** [comment]
> @johnmutuma5 the title of your PR sort of conflicts with the section of the description that says `What does this PR do?`. The title tells me that you're approving multi-city requests while the description tells me you're making sure managers can see trips associated with a multi-city request.

@d3mola, I have updated the title, I hope it goes well with the description now.

File: src/modules/requests/RequestsController.js [hunk 3]
```diff
@@ -8,12 +8,14 @@ import ApprovalsController from '../approvals/ApprovalsController';
...
   static async createRequest(req, res) {
     try {
+      const { trips } = req.body;
       const requestData = {
         ...req.body,
         id: Utils.generateUniqueId(),
         userId: req.user.UserInfo.id
       };
       const newRequest = await models.Request.create(requestData);
+      await Promise.all(trips.map(trip => newRequest.createTrip(trip)));
```
[the remark this answers is not in the document]
Comment (johnmutuma5) [reply src/modules/requests/RequestsController.js]: @d3mola, not really, this was just a quick solution to help with testing and demonstrating the functionality. @Spectrumsun, @amarachukwu-agbo  and @seunzone  will be incorporating a fully fledged and more comprehensive logic for the creation of requests. Perhaps I should have included a `FIXME` comment there in...

**Pull request merged**